File: 天梯赛/L2-008. 最长对称子串.cpp
#include<iostream>
#include<string>
#include<algorithm>
using namespace std;
int main(){
	string str,a,b;
	getline(cin,str);
	int maxlen=str.length(),res=0;
	for(int i=0;i<maxlen;++i){//curr index
		int tmp=1;
		for(int j=1;j+i<maxlen&&j<=i;++j){//substr(i,j)
			if(str[i-j]!=str[i+j])
				break;
			tmp+=2;
		}
		res=max(res,tmp);
		tmp=0;
		for(int j=1;j+i<maxlen&&j<=i+1;++j){
			if(str[i-j+1]!=str[i+j])
				break;
			tmp+=2;
		}
		res=max(res,tmp);
	}
	cout<<res;
	return 0;
}

File: 天梯赛/java/L2-008. 最长对称子串.java
import java.util.Scanner;

public class Main {
	
	public static void main(String[] args) {
		Scanner in=new Scanner(System.in);
		String str=in.nextLine();
		int maxlen=str.length(),res=0;
		for(int i=0;i<maxlen;++i) {
			int tmp=1;
			for(int j=1;j+i<maxlen&&j<=i;++j) {
				if(str.charAt(i-j)!=str.charAt(i+j))break;
				tmp+=2;
			}
			res=Math.max(res, tmp);
			tmp=0;
			for(int j=1;j+i<maxlen&&j<=i+1;++j) {
				if(str.charAt(i-j+1)!=str.charAt(i+j))break;
				tmp+=2;
			}
			res=Math.max(res, tmp);
		}
		System.out.print(res);
	}
	
}

